### PR TITLE
Upgrade nginx to 1.26.1 since it is the new stable version

### PR DESCRIPTION
**Pull request by tgxworld**
Since 1.26.x has been marked stable, 1.25.x is automatically EOL as
nginx only maintains one mainline version and one stable version.

### CODE REVIEW

File: image/base/install-nginx [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
-VERSION=1.25.3
-HASH="64c5b975ca287939e828303fa857d22f142b251f17808dfe41733512d9cded86"
+VERSION=1.26.1
 
 cd /tmp
 wget -q https://nginx.org/download/nginx-$VERSION.tar.gz
-sha256sum nginx-$VERSION.tar.gz
-echo "$HASH nginx-$VERSION.tar.gz" | sha256sum -c
+wget -q https://nginx.org/download/nginx-$VERSION.tar.gz.asc
+gpg --import nginx-$VERSION.tar.gz.asc
```
Comment (saj) [inline image/base/install-nginx:10]: `gpg --import` will import one or more PGP public keys into the GnuPG key ring.  This `.asc` file is not a public key -- it is a signature.

```
% gpg --import nginx-1.26.1.tar.gz.asc
gpg: no valid OpenPGP data found.
gpg: Total number processed: 0
% echo $?
2
```

Release keys are published here.
https://nginx.org/en/pgp_keys.html

```
% curl -Ssf https://nginx.org/keys/nginx_signing.key | gpg --import
gpg: key 2FD21310B49F6B46: 4 signatures not checked due to missing keys
gpg: key 2FD21310B49F6B46: public key "nginx signing key <signing-key-2@nginx.com>" imported
gpg: key ABF5BD827BD9BF62: 6 signatures not checked due to missing keys
gpg: key ABF5BD827BD9BF62: public key "nginx signing key <signing-key@nginx.com>" imported
gpg: key BCDCD8A38D88A2B3: 3 signatures not checked due to missing keys
gpg: key BCDCD8A38D88A2B3: public key "nginx signing key <signing-key-3@nginx.com>" imported
gpg: Total number processed: 3
gpg:               imported: 3
gpg: no ultimately trusted keys found
% echo $?
0
```

The line below this one makes sense.  Once the release keys are imported, you may verify the signature in the `.asc` file.

File: image/base/pluknet.key [hunk 2]
```diff
@@ -0,0 +1,65 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
```
Comment (tgxworld) [inline image/base/pluknet.key:1]: See https://github.com/discourse/discourse_docker/pull/827/files#diff-fa8cf8fc82082ca028fdbc192cdfc37c755b819c68eb457c796e3b713e73c5a6R79-R80

File: image/base/nginx-1.26.1.tar.gz.asc [hunk 3]
```diff
@@ -0,0 +1,17 @@
+-----BEGIN PGP SIGNATURE-----
```
Comment (tgxworld) [inline image/base/nginx-1.26.1.tar.gz.asc:1]: Downloaded from https://nginx.org/en/download.html

File: image/base/pluknet.key (hunk 2 above)
Comment (tgxworld) [reply image/base/pluknet.key:1]: Obtained from https://nginx.org/en/pgp_keys.html

File: image/base/install-nginx (hunk 1 above)
Comment (tgxworld) [reply image/base/install-nginx]: Thank you for the tip here 👍 I've applied your feedback into the PR.

File: image/base/slim.Dockerfile [hunk 4]
```diff
@@ -74,6 +74,12 @@ ADD install-jemalloc /tmp/install-jemalloc
 RUN /tmp/install-jemalloc
 
 ADD install-nginx /tmp/install-nginx
+ADD nginx-1.26.1.tar.gz.asc /tmp/nginx-1.26.1.tar.gz.asc
+
+# From https://nginx.org/en/pgp_keys.html and nginx doesn't seem to indicate whose public key was used to sign the package
+# so there is some trial and error required here to figure out the right public key.
+ADD pluknet.key /tmp/nginx_public.key
```
Comment (CvX) [inline image/base/slim.Dockerfile:81]: So `nginx_signing.key` from that pgp_keys page isn't the key used to sign nginx? 😅

Comment (tgxworld) [reply image/base/slim.Dockerfile:81]: It is apparently used to sign the OS packages and not the files from the download page.

**Review by CvX — APPROVED** [review]

File: image/base/nginx-1.26.1.tar.gz.asc (hunk 3 above)
Comment (saj) [reply image/base/nginx-1.26.1.tar.gz.asc:1]: It is safe to retrieve this signature from the web if you would find that to be more convenient.  (Consider that when this is next bumped, the signature will change.)

Ensuring trust on the key is of sole importance.
With the key vendored, we can use that to verify everything else.
(As a bonus, the key -- or keys -- should not change between releases.)

File: image/base/slim.Dockerfile (hunk 4 above)
Comment (saj) [reply image/base/slim.Dockerfile:81]: Oh, this confused me as well.

I guess _pluknet_ is Sergey Kandaurov's handle.  Would it help to name the file accordingly?  Perhaps its purpose would be more obvious.

```
image/base/nginx-dev-pluknet.key
```

or

```
image/base/nginx-sergey-kandaurov.key
```

(The file name has no significance to PGP.)

Three other developer keys are listed on that page.  If it is unclear who will be signing future releases, we could vendor and `gpg --import` all four keys.  On `--verify`, `gpg` will pass if it is able to verify a signature against any key.  In other words, this procedure will remain forwards compatible so long as any one of those four people sign a future release.

**Review by saj — APPROVED** [review]
LGTM but see comments.

**Comment by tgxworld** [comment]
Thank you @saj and @CvX for reviewing here 🙇

**Pull request merged**